lib/Interpreter/CppInterOp.cpp

```diff
@@ -1128,9 +1128,33 @@ namespace Cpp {
 
     if (auto *CXXRD = llvm::dyn_cast_or_null<CXXRecordDecl>(D)) {
       std::vector<TCppScope_t> datamembers;
-      for (auto it = CXXRD->field_begin(), end = CXXRD->field_end(); it != end;
-           it++) {
-        datamembers.push_back((TCppScope_t)*it);
+      llvm::SmallVector<RecordDecl::field_iterator, 2> stack_begin;
+      llvm::SmallVector<RecordDecl::field_iterator, 2> stack_end;
+      stack_begin.push_back(CXXRD->field_begin());
+      stack_end.push_back(CXXRD->field_end());
+      while (!stack_begin.empty()) {
+        if (stack_begin.back() == stack_end.back()) {
+          stack_begin.pop_back();
+          stack_end.pop_back();
+          continue;
+        }
+        Decl* D = *(stack_begin.back());
+        if (D->getKind() == clang::Decl::Field &&
+            dyn_cast<FieldDecl>(D)->isAnonymousStructOrUnion()) {
```

Comment:
```suggestion
        if (isa<FieldDecl>(D) &&
            cast<FieldDecl>(D)->isAnonymousStructOrUnion()) {
```

Reply:
I think we should have a proper dyn_cast and `isAnonymousStructOrUnion` as a nested if...